src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -16,74 +16,69 @@
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
 import com.google.sps.data.User;
+import java.util.Map;
+import java.util.HashMap;
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
     UserService userService = UserServiceFactory.getUserService();
 
     String urlToRedirect = "/";
     String userId = "null";
     String logoutUrl = "null";
     String loginUrl = "null";
     String userEmail = "null";
-
-    // Boolean to update User's main page 
     Boolean showNewTab = false;
 
+    User user;
     if (userService.isUserLoggedIn()) {
-      userEmail = userService.getCurrentUser().getEmail();
       userId = userService.getCurrentUser().getUserId();
+      userEmail = userService.getCurrentUser().getEmail();
       logoutUrl = userService.createLogoutURL(urlToRedirect);
       showNewTab = true;
     } else {
       loginUrl = userService.createLoginURL(urlToRedirect);
     }
 
-    User user = new User(userId, loginUrl, logoutUrl, userEmail, showNewTab);
-    
-    if (userId != "null") {
-        checkUserInDatastore(user);
+    user = new User(userId, userEmail);
+
+    if (userId != "null" && !isUserInDatastore(user)) {
+      storeUserToDatastore(user);
```

Comment:
are there circumstances when the user exists and is logged in but they are not in the datastore?